sycl/test-e2e/Basic/interop/interop_all_backends.cpp

```diff
@@ -0,0 +1,64 @@
+// REQUIRES: CUDA || HIP
+// RUN: %{build} %if hip %{ -DSYCL_EXT_ONEAPI_BACKEND_HIP %} %else %{ %if cuda %{ -DSYCL_EXT_ONEAPI_BACKEND_CUDA_EXPERIMENTAL %} %else %{ %if level_zero %{ -DSYCL_EXT_ONEAPI_BACKEND_L0 %} %} %} -o %t.out
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+#ifdef SYCL_EXT_ONEAPI_BACKEND_CUDA_EXPERIMENTAL
+#include <sycl/ext/oneapi/experimental/backend/cuda.hpp>
+constexpr auto BACKEND = backend::ext_oneapi_cuda;
+#elif defined(SYCL_EXT_ONEAPI_BACKEND_HIP)
+#include <sycl/ext/oneapi/backend/hip.hpp>
+constexpr auto BACKEND = backend::ext_oneapi_hip;
+#elif defined(SYCL_EXT_ONEAPI_BACKEND_L0)
+constexpr auto BACKEND = backend::ext_oneapi_level_zero;
+#else
+constexpr auto BACKEND = backend::opencl;
+#endif
+
+constexpr int N = 100;
+constexpr int VAL = 3;
+
+int main() {
+
+  device Device;
+  auto NativeDevice = get_native<BACKEND>(Device);
```

Comment:
Instead of using auto, could we consider making backend-dependent type aliases for these? I believe the backend specifications should be making some requirement for what they are, so it would be another level of checks we would do.

Reply:
Sure,
Do you mean having backend dependent alias' for e.g.

`TYPE NativeDevice`
such that `TYPE == CUdevice` for cuda etc?
Or do you mean using
`backend_traits<backend::ext_oneapi_cuda>::return_type<device> NativeDevice`

where `backend_traits<backend::ext_oneapi_cuda>::return_type<device> ==CUdevice`?